classes/3.html

```diff
@@ -199,22 +263,81 @@ <h3>XHR</h3>
             xhr.send();
           </code></pre>
         </section>
+      </section>
+      <!-- Promesas y fetch API -->
+      <section>
+        <h3>Promesas</h3>
+        <section>
+          Una <a href="https://developer.mozilla.org/es/docs/Web/JavaScript/Referencia/Objetos_globales/Promise">promesa</a> es un objeto que representa el estado de una operación asíncrona. El navegador nos promete retornar una respuesta cuando la reciba, si la recibe
+        </section>
         <section>
-          <h3>fetch</h3>
-          Fetch hace más fácil el manejo de peticiones y respuestas del servidor, en comparación con XHR.
-          <pre><code data-trim class="lang-javascript">
-            fetch(
-              "https://gist.githubusercontent.com/spereztoledo/d7740ac1ad1212ce8810c5c809d7acd3/raw/d57477f19d8e173b8671ffff9494db75e246d4e7/quotes.json"
-            )
-              .then((data) => {
-                console.log(data);
+          <h3> Estados de una promesa</h3>
+            <h4 style="color:blue">pending</h4>
+            <p>Estado inicial, ni completada ni rechazada</p>
+            <h4 style="color:green">fulfilled</h4>
+            <p>Completada satisfactoriamente, retornando un valor</p>
+            <h4 style="color:red">rejected</h4>
+            <p>La operación falló, retornando un error</p>
+        </section>
+        <section>
+          <p>Para usar una promesa, se encadenan metodos, que ejecutarán callbacks según lo que suceda con la respuesta. Estos callbacks recibirán los datos que podemos manejar dentro de estas funciones. Los metodos son:</p>
+          <ul>
+            <li><a href="https://developer.mozilla.org/es/docs/Web/JavaScript/Referencia/Objetos_globales/Promise/then">then</a></li>
+            <li><a href="https://developer.mozilla.org/es/docs/Web/JavaScript/Referencia/Objetos_globales/Promise/catch">catch</a></li>
+            <li><a href="https://developer.mozilla.org/es/docs/Web/JavaScript/Referencia/Objetos_globales/Promise/finally">finally</a></li>
+          </ul>
+        </section>
+        <section>
+          Un buen ejemplo de uso de promesas es la API <a href="https://developer.mozilla.org/es/docs/Web/API/Fetch_API">fetch</a>. Esta facilita el manejo de peticiones y respuestas a un servidor, en comparación con XHR.
+          <pre><code data-trim data-line-numbers="1|2|3|7|8|9|10|11|1-12" class="lang-javascript">
+            const promesa = fetch("https://api.ejemplo.com/datos");
+            promesa.then(function (data) {
                 return data.json();
```

Comment:
Creo que en esta parte va a ser muy importante que agreguemos algunas notas. Capaz podamos poner algunos comentarios en esta parte del código sobre qué es data, que hace la instrucción `data.json()`, que pasa con la promesa que se devuelve en ese caso

Reply:
Si, acá creo que va a venir bien tener algunas diapos más y hacer la introducción más lenta, sino queda todo muy de golpe. Igualmente, si ves los `data-line-numbers` la idea is irse deteniendo casi que linea a linea para analizarlas y contar que está pasando, pero también faltan las notas sobre eso.

Reply:
Excelente! Si, vi lo de las líneas y coincido, con unas notas que acompañen estamos 💯